app/controllers/UsersController.scala

```diff
@@ -126,4 +161,91 @@ class UsersController @Inject()(messagesApi: MessagesApi,
     Redirect(routes.HomeController.index()).withNewSession
   }
 
+  def manageUser(pageNumber: Int = 1, keyword: Option[String] = None): Action[AnyContent] = adminAction.async { implicit request =>
+    usersRepository
+      .paginate(pageNumber, keyword)
+      .flatMap { userInfo =>
+        val users = userInfo map (user =>
+          ManageUserInfo(user.email, user.active, user._id.stringify))
+
+        usersRepository
+          .userCountWithKeyword(keyword)
+          .map { count =>
+            val pages = Math.ceil(count / 10D).toInt
+
+            Ok(views.html.users.manageusers(users, pages, pageNumber, keyword))
+          }
+      }
+  }
+
+  def searchUser: Action[AnyContent] = adminAction.async { implicit request =>
+    emailForm.bindFromRequest.fold(
+      formWithErrors => {
+        Logger.error(s"Received a bad request for user manage ==> $formWithErrors")
+        Future.successful(BadRequest(" OOps! Invalid value encountered !"))
+      },
+      userInformation => {
+        usersRepository
+          .paginate(userInformation.page, userInformation.email)
+          .flatMap { userInfo =>
+            val users = userInfo map (user =>
+              ManageUserInfo(user.email, user.active, user._id.stringify))
+
+            usersRepository
+              .userCountWithKeyword(userInformation.email)
+              .map { count =>
+                val pages = Math.ceil(count / 10D).toInt
+
+                Ok(Json.toJson(UserSearchResult(users, pages, userInformation.page, userInformation.email.getOrElse(""))).toString)
+              }
+          }
+      }
+    )
+  }
+
+  def updateUser(): Action[AnyContent] = adminAction.async { implicit request =>
+    updateUserForm.bindFromRequest.fold(
+      formWithErrors => {
+        Logger.error(s"Received a bad request for user manage $formWithErrors")
+        Future.successful(BadRequest(views.html.users.updateuser(formWithErrors)))
+      },
+      userInfo => {
+        usersRepository.update(UpdatedUserInfo(userInfo.email, userInfo.active, userInfo.password))
+          .flatMap { result =>
+            if (result.ok) {
+              Logger.info(s"User details successfully updated for ${userInfo.email}")
+              Future.successful(Redirect(routes.UsersController.manageUser(1, None))
+                .flashing("message" -> s"Details successfully updated for ${userInfo.email}"))
+            }
+            else {
+              Future.successful(InternalServerError("Something went wrong!"))
+            }
+          }
+      })
+  }
+
+  def update(email: String): Action[AnyContent] = adminAction.async { implicit request =>
```

Comment:
If this is getting user for update page, renaming it to getByEmail would make more sense.